spline-shadow/client-ui/src/app/components/lineage/lineage-details/schema-details/schema-details.component.ts

```diff
@@ -42,17 +53,105 @@ export class SchemaDetailsComponent implements OnInit {
     return this.lineageGraphService.getColorFromOperationType(this.getType())
   }
 
-  getType(): string {
+  getType(property?: any): string {
+    if (property) {
+      return property._typeHint ? property._typeHint.split('.').pop() : undefined
+    }
     return this.detailsInfo._typeHint.split('.').pop()
   }
 
-  getExpression(): string {
+  getExpressions(): Expression[] {
+    let expressions = []
     switch (this.getType()) {
       case OperationType.Join:
-        return "<div class=expression>" + this.detailsInfo.joinType + " join on</div><code>" + this.detailsInfo.condition.text + "</code>"
-      default:
-        //TODO : Implement the other expressions. 
-        return "..."
+        // Build the join expression
+        const title = this.detailsInfo.joinType + " join on"
+        const values = ["<code>" + this.detailsInfo.condition.text + "</code>"]
+        const expression = new Expression(title, values)
+        expressions.push(expression)
+        break
+      case OperationType.Projection:
+        // Build the transformations expressions
+        if (this.detailsInfo.transformations) {
+          const title = "Transformations"
+          const values = new Array()
+          _.each(this.detailsInfo.transformations, transformation => values.push("<i class='transformation-prexif'>λ = </i><code>" + this.getText(transformation) + "</code>"))
```

Comment:
Is it possible to use some templating mechanism to avoid concatenating string literals with hardcoding html tags?